tests/v2/validation/fleet/public_repo_test.go

```diff
@@ -0,0 +1,111 @@
+//go:build validation || airgap
+
+package fleet
+
+import (
+	"testing"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extensionscluster "github.com/rancher/shepherd/extensions/clusters"
+	extensionsfleet "github.com/rancher/shepherd/extensions/fleet"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FleetPublicRepoTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	provisioningConfig *provisioninginput.Config
+}
+
+func (f *FleetPublicRepoTestSuite) TearDownSuite() {
+	f.session.Cleanup()
+}
+
+func (f *FleetPublicRepoTestSuite) SetupSuite() {
+	f.session = session.NewSession()
+
+	client, err := rancher.NewClient("", f.session)
+	require.NoError(f.T(), err)
+
+	f.client = client
+
+	userConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+
+	f.provisioningConfig = userConfig
+}
+
+func (f *FleetPublicRepoTestSuite) TestGitRepoDeployment() {
+	defer f.session.Cleanup()
+
+	var clusterID string
+	var err error
+	clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(f.client, f.client.RancherConfig.ClusterName, "fleet-default")
+	if clusterObject != nil {
+		status := &provv1.ClusterStatus{}
+		err := steveV1.ConvertToK8sType(clusterObject.Status, status)
+		require.NoError(f.T(), err)
+
+		clusterID = status.ClusterName
+		logrus.Info(clusterID)
+	} else {
+		clusterID, err = extensionscluster.GetClusterIDByName(f.client, f.client.RancherConfig.ClusterName)
+		require.NoError(f.T(), err)
+	}
+
+	podErrors := pods.StatusPods(f.client, clusterID)
+	require.Empty(f.T(), podErrors)
+
+	fleetVersion, err := getDeploymentVersion(f.client, fleetControllerName, "local")
+	require.NoError(f.T(), err)
+
+	rancherVersion, err := getDeploymentVersion(f.client, "cattle-system/rancher", "local")
+	require.NoError(f.T(), err)
+
+	// not sure if we want this but may be nice to have the rancher version in the test name, but will cause bloat in qase
+	f.Run("rancher "+rancherVersion+"/fleet "+fleetVersion, func() {
+		fleetGitRepo := v1alpha1.GitRepo{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "automatedrepo-" + namegenerator.RandStringLower(5),
+				Namespace: "fleet-default",
+			},
+			Spec: v1alpha1.GitRepoSpec{
+				Repo:                "https://github.com/rancher/fleet-examples", // "https://github.com/rancher/fleet-examplestypo",
```

Comment:
instead of hard coding some of these values, how about we create a config for fleet, where in user has an option to send in custom git repo, path and branch. that way we can verify other deployments as well easily.

Reply:
this is a good idea for a dynamic test. But this test is designed to be static for all these variables. That way we can run it for release testing without providing a config.

Reply:
For release testing, I agree with static values, its easier to identify any issues that we run into without over complicating the config. However, if we have the test to support a config from the start could future-proof the test by allowing flexibility.  especially if we want to change things like the Git repo or branch without touching the core test logic..

let me know your thoughts on this.

Reply:
can this be added in a separate PR since its outside the scope of the linked ticket?

Reply:
https://github.com/rancher/qa-tasks/issues/1530